### PR TITLE
Add docs for datamodule hparams

### LINKED ISSUE #9102: Add docs for datamodule hparams
## 📚 Documentation

Add missing docs for #3792

(Possibly depends on #8948 if there are changes)

### PR DESCRIPTION
**Pull request by s-rog**
## What does this PR do?
Fixes #9102 (which adds missing docs for #3792)

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

## Notes
Perhaps the bottom is not the best place for this?
[preview](https://129007-178626720-gh.circle-artifacts.com/0/html/extensions/datamodules.html#hyperparameters-in-datamodules)

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]

**Comment by akihironitta** [comment]
For easy access to the built docs: https://129007-178626720-gh.circle-artifacts.com/0/html/extensions/datamodules.html#hyperparameters-in-datamodules

File: docs/source/extensions/datamodules.rst [hunk 1]
```diff
@@ -432,3 +432,21 @@ You can of course use DataModules in plain PyTorch code as well.
...
+Hyperparameters in DataModules
+------------------------------
+Like LightningModules, DataModules support hyperparameters with the same API.
+
+.. code-block:: python
+
+    import pytorch_lightning as pl
+
+
+    class CustomDataModule(pl.LightningDataModule):
```
Comment (awaelchli) [inline docs/source/extensions/datamodules.rst:447]:
```suggestion
.. code-block:: python

    import pytorch_lightning as pl

    class CustomDataModule(pl.LightningDataModule):
```

File: docs/source/extensions/datamodules.rst [hunk 2]
```diff
@@ -432,3 +432,21 @@ You can of course use DataModules in plain PyTorch code as well.
...
+        def __init__(self, *args, **kwargs):
+            super().__init__()
+            self.save_hyperparameters()
+
+Refer to `save_hyperparameters` in :doc:`lightning module <../common/lightning_module>` for more details.
```
Comment (awaelchli) [inline docs/source/extensions/datamodules.rst:452]:
```suggestion
Refer to :meth:`~pytorch_lightning.core.lightning.LightningModule.save_hyperparameters` for more details.
```
Can you check if linking to it directly would work?

**Review by awaelchli — APPROVED** [review]

File: docs/source/extensions/datamodules.rst (hunk 2 above)
Comment (s-rog) [reply docs/source/extensions/datamodules.rst:451]: it only highlights unfortunately

**Review by kaushikb11 — APPROVED** [review]
